plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,614 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: s3_object_info
+version_added: 5.0.0
+short_description: Gather information about objects in S3
+description:
+    - Describes objects in S3.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  bucket_name:
+    description:
+      - The name of the bucket that contains the object.
+    required: true
+    type: str
+  object_name:
+    description:
+      - The name of the object.
+    required: false
+    type: str
+  object_attributes:
+    description:
+      - The fields/details that should be returned.
+      - Required when I(object_attributes) is C(true) in I(object_details).
```

Comment:
(non-blocking)

I'd be tempted to default this to returning the 5 current attributes.